backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceBreadcrumbs.java

```diff
@@ -0,0 +1,342 @@
+package backtraceio.library.breadcrumbs;
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.Context;
+import android.support.annotation.NonNull;
+
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import backtraceio.library.base.BacktraceBase;
+import backtraceio.library.enums.BacktraceBreadcrumbLevel;
+import backtraceio.library.enums.BacktraceBreadcrumbType;
+import backtraceio.library.events.OnServerResponseEventListener;
+import backtraceio.library.interfaces.Breadcrumbs;
+import backtraceio.library.logger.BacktraceLogger;
+import backtraceio.library.models.BacktraceData;
+import backtraceio.library.models.BacktraceResult;
+import backtraceio.library.models.database.BacktraceDatabaseRecord;
+import backtraceio.library.models.json.BacktraceReport;
+
+public class BacktraceBreadcrumbs implements Breadcrumbs {
+
+    private static transient String LOG_TAG = BacktraceBreadcrumbs.class.getSimpleName();
+
+    /**
+     * Which breadcrumb types are enabled?
+     */
+    private Set<BacktraceBreadcrumbType> enabledBreadcrumbTypes;
+
+    /**
+     * The Backtrace BroadcastReciever instance
+     */
+    private BacktraceBroadcastReceiver backtraceBroadcastReceiver;
+
+    /**
+     * The Backtrace ComponentCallbacks2 listener
+     */
+    private BacktraceComponentListener backtraceComponentListener;
+
+    /**
+     * The Backtrace ActivityLifecycleCallbacks listener
+     */
+    private BacktraceActivityLifecycleListener backtraceActivityLifecycleListener;
+
+    /**
+     * The Backtrace Breadcrumbs logger instance
+     */
+    private BacktraceBreadcrumbsLogger backtraceBreadcrumbsLogger;
+
+    final private static String breadcrumbLogDirectory = "breadcrumbs";
```

Comment:
I want this variable to be static so I can call it from a static function `BacktraceBreadcrumbs::getBreadcrumbsLogPath`. I need this to initialize Crashpad attachment paths since we might not have a `BacktraceBreadcrumbs` instance yet but Crashpad needs the attachment paths when it initiailizes.

Reply:
No longer relevant after your refactor suggestion...it did make things much cleaner. Thank you!